website/src/word-panel.tsx

```diff
@@ -315,32 +322,98 @@ const EditWordPartGloss = (props: {
   index: number
   options: GroupedOption[]
 }) => {
+  const preferences = usePreferences()
   const { form } = useForm()
+  const [, insertCustomMorphemeTag] = useMutation(
+    Dailp.InsertCustomMorphemeTagDocument
+  )
+  const [{ data: allTagsData }, executeQuery] = useQuery({
+    query: Dailp.GlossaryDocument,
+    variables: { system: preferences.cherokeeRepresentation },
+  })
+
+  let allNewTags = allTagsData?.allTags.filter(
+    (tag: Dailp.MorphemeTag) => tag.tag !== ""
+  )
+  const groupedTags = groupBy(allNewTags, (t) => t.morphemeType)
+
+  // Creates a selectable option out of each functional tag, and groups them together by morpheme type.
+  const allNewOptions: GroupedOption[] = Object.entries(groupedTags).map(
+    ([group, tags]) => {
+      return {
+        label: group,
+        options: tags.map((tag) => {
+          return {
+            // Value is a custom type to track data of a morpheme's gloss in its string form and its matching tag, if there is one.
+            value: tag.tag,
+            label: tag.title,
+          }
+        }),
+      }
+    }
+  )
 
   // Handles gloss selection and creation of new glosses.
-  const handleChange = (
+  const handleNewTag = async (
```

Comment:
We don't have official guidance on async/await at this time but we tend to use callbacks elsewhere.

That being said, since there are no guidelines this is fine.